graph/tags.go

```diff
@@ -222,6 +223,15 @@ func (store *TagStore) Set(repoName, tag, imageName string, force bool) error {
 	var repo Repository
 	repoName = registry.NormalizeLocalName(repoName)
 	if r, exists := store.Repositories[repoName]; exists {
+
+		if id, exists := store.Repositories[repoName][tag]; exists {
+
+			if id != img.ID && job != nil && job.Name == "load" {
+				str := "The image " + repoName + ":" + tag + " already exists, renaming the old one with id " + id[:12] + " to <none>:<none>\n"
+				job.Stdout.Write([]byte(str))
+			}
+		}
+
 		repo = r
 		if old, exists := store.Repositories[repoName][tag]; exists && !force {
```

Comment:
If you are talking about line 227, this one is the existing upstream code.
If you like I can change this to r[tag] instead of store.Repositories[repoName][tag] and see if that works.